toml/src/main/kotlin/org/rust/toml/intentions/SimplifyDependencySpecificationIntention.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import com.intellij.psi.util.parentsOfType
+import org.rust.cargo.CargoConstants
+import org.rust.lang.core.psi.ext.endOffset
+import org.rust.toml.isDependencyListHeader
+import org.toml.lang.psi.TomlInlineTable
+import org.toml.lang.psi.TomlKeyValue
+import org.toml.lang.psi.TomlPsiFactory
+import org.toml.lang.psi.TomlTable
+
+class SimplifyDependencySpecificationIntention : TomlElementBaseIntentionAction<TomlKeyValue>() {
+    override fun getText() = "Simplify dependency specification"
+    override fun getFamilyName(): String = text
+
+    override fun findApplicableContextInternal(project: Project, editor: Editor, element: PsiElement): TomlKeyValue? {
+        if (element.containingFile.name != CargoConstants.MANIFEST_FILE) return null
+
+        val dependencyTable = element.parentOfType<TomlTable>()
+            ?.takeIf { it.header.isDependencyListHeader }
+            ?: return null
+
+        val dependency = element.parentsOfType<TomlKeyValue>()
+            .firstOrNull { it.parent == dependencyTable }
+            ?: return null
+
+        val dependencyValue = dependency.value
+        return dependency.takeIf {
+            dependencyValue is TomlInlineTable &&
+                dependencyValue.entries.singleOrNull()?.key?.segments?.singleOrNull()?.text == "version"
+        }
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: TomlKeyValue) {
+        val crateName = ctx.key.text
+        val crateVersion = (ctx.value as? TomlInlineTable)?.entries?.singleOrNull { it.key.text == "version" }?.value?.text ?: return
+        val newKeyValue = TomlPsiFactory(project).createKeyValue("$crateName = $crateVersion")
+        val newKeyValueOffset = ctx.replace(newKeyValue).endOffset
```

Comment:
I wonder if it would be better to replace only property value as it is gets changed only

Reply:
Yep, it even looks simpler: 
```
val version = (ctx.value as? TomlInlineTable)?.entries?.singleOrNull { it.key.text == "version" }?.value ?: return
ctx.value.replace(version.copy())
```